src/Tools/dotnet-monitor/Egress/AzureBlob/AzureBlobEgressProvider.cs

```diff
@@ -129,6 +157,91 @@ private Uri GetAccountUri(AzureBlobEgressProviderOptions options, out string acc
             return blobUriBuilder.ToUri();
         }
 
+        private Uri GetQueueAccountUri(AzureBlobEgressProviderOptions options, out string accountName)
+        {
+            var queueUriBuilder = new QueueUriBuilder(options.QueueAccountUri);
+
+            queueUriBuilder.Query = null;
+            queueUriBuilder.QueueName = null;
+
+            accountName = queueUriBuilder.AccountName;
+
+            return queueUriBuilder.ToUri();
+        }
+
+        public async Task EgressMessageToQueue(string artifactName, AzureBlobEgressProviderOptions options, CancellationToken token)
+        {
+            try
+            {
+                QueueClient queueClient = await GetQueueClientAsync(options, token);
+
+                if (queueClient.Exists())
```

Comment:
This code path silently fails to write to the queue because the queue doesn't exist. Because we can't upload the blob storage and push to the queue as an atomic operation and there's potential benefit to allowing the egress operation to complete even if it can't write to the queue, I think this is okay.

However, if the queue doesn't exist, we should log a warning to that effect so that the user or operator know why the blob appeared but the message was not queued. In that same vein, we shouldn't fail the whole operation if there is a problem in writing to the queue either; we should handle the exception and log it appropriately, probably as a warning.

To sum up, either all types of failures should fail the operation or all types of failures should be handled and logged; I prefer the latter.